supervisor/resolution/check.py

```diff
@@ -31,14 +29,21 @@ def data(self) -> Dict[str, Any]:
     @property
     def all_checks(self) -> List[CheckBase]:
         """Return all list of all checks."""
-        return [self._core_security, self._free_space, self._addon_pwned]
+        return list(self._checks.values())
+
+    def _load(self):
+        """Load all checks."""
+        package = f"{__package__}.checks"
+        for module in get_valid_modules("checks"):
+            check_module = import_module(f"{package}.{module}")
+            check = check_module.setup(self.coresys)
+            self._checks[check.slug] = check
```

Comment:
```suggestion
            self._checks[check.slug] = check_module.setup(self.coresys)
```
No need to define `check` here